src/node_api.cc

```diff
@@ -1409,7 +1409,39 @@ napi_status napi_create_symbol(napi_env env,
   return GET_RETURN_STATUS(env);
 }
 
+static napi_status set_error_code(napi_env env,
+                                  napi_value code,
+                                  const char* code_cstring,
+                                  v8::Local<v8::Value> val) {
+  if ((code != nullptr) || (code_cstring != nullptr)) {
+    v8::Isolate* isolate = env->isolate;
+    v8::Local<v8::Context> context = isolate->GetCurrentContext();
+
+    v8::MaybeLocal<v8::Object> maybe_obj = val->ToObject(context);
```

Comment:
I don't think we even need the IsObject() check since for all of the callers we know we are passing in an Object so that does make the code more compact.